### DISCUSSION TITLE
Error in custom defined funtion

### DISCUSSION THREAD

**Discussion by SP171995** [pybamm-team/PyBaMM#2815] [Q&A]
python version 3.8.3
pybamm version 23.2

I have defined a new function for electrode diffusivity as function of c_s_surf and using lookup table for concentration and diffusion coefficients : 

def solid_state_diff_graphite (c_s_surf) :
    print(c_s_surf)
    x = np.array([0,...,1])
    y = np.array([])
    D_s_graphite = pybamm.Interpolant(x, y, c_s_surf, interpolator='linear')
    return D_s_graphite

def solid_state_diff_nmc (c_s_surf) :
    x = np.array([0,...,1])
    y = np.array([.....])
    D_s_nmc = pybamm.Interpolant(x, y, c_s_surf, interpolator='linear')
    return D_s_nmc

but I am getting an error as follows
TypeError: solid_state_diff_graphite() takes 1 positional argument but 2 were given

i also tried using self 
def solid_state_diff_nmc (self,c_s_surf) :

but this is not calling c_s_surf insted it is taking it as Reference Temperature and interpolation is failing

**Comment by valentinsulzer** [comment]
Look at existing parameter sets to see which inputs are required

**Reply by SP171995** [reply]
so existing parameter set just has one value of diffusion coefficient but I have defined it now as
"Negative electrode diffusivity [m2.s-1]": solid_state_diff_graphite,
"Positive electrode diffusivity [m2.s-1]": solid_state_diff_nmc,

if I define the function as "def solid_state_diff_graphite (self,c_s_surf) : "
then code is taking c_s_surf as scalar value 298.15

**Reply by valentinsulzer** [reply]
https://github.com/pybamm-team/PyBaMM/blob/develop/pybamm/input/parameters/lithium_ion/OKane2022.py look at how it's defined here
